geoapps/inversion/potential_fields/magnetic_scalar/application.py

```diff
@@ -0,0 +1,135 @@
+            # Regularization
+            State(component_id="alpha_s", component_property="value"),
+            State(component_id="length_scale_x", component_property="value"),
+            State(component_id="length_scale_y", component_property="value"),
+            State(component_id="length_scale_xz", component_property="value"),
+            State(component_id="s_norm", component_property="value"),
+            State(component_id="x_norm", component_property="value"),
+            State(component_id="y_norm", component_property="value"),
+            State(component_id="z_norm", component_property="value"),
+            # Upper-Lower Bounds
+            State(component_id="lower_bound_options", component_property="value"),
+            State(component_id="lower_bound_data", component_property="value"),
+            State(component_id="lower_bound_const", component_property="value"),
+            State(component_id="upper_bound_options", component_property="value"),
+            State(component_id="upper_bound_data", component_property="value"),
+            State(component_id="upper_bound_const", component_property="value"),
+            # Detrend
+            State(component_id="detrend_type", component_property="value"),
+            State(component_id="detrend_order", component_property="value"),
+            # Ignore Values
+            State(component_id="ignore_values", component_property="value"),
+            # Optimization
+            State(component_id="max_global_iterations", component_property="value"),
+            State(component_id="max_irls_iterations", component_property="value"),
+            State(component_id="coolingRate", component_property="value"),
+            State(component_id="coolingFactor", component_property="value"),
+            State(component_id="chi_factor", component_property="value"),
+            State(component_id="initial_beta_ratio", component_property="value"),
+            State(component_id="max_cg_iterations", component_property="value"),
+            State(component_id="tol_cg", component_property="value"),
+            State(component_id="n_cpu", component_property="value"),
+            State(component_id="store_sensitivities", component_property="value"),
+            State(component_id="tile_spatial", component_property="value"),
+            # Output
+            State(component_id="ga_group", component_property="value"),
+            State(component_id="monitoring_directory", component_property="value"),
+            # Magnetic specific
+            State(component_id="inducing_field_strength", component_property="value"),
+            State(
+                component_id="inducing_field_inclination", component_property="value"
+            ),
+            State(
+                component_id="inducing_field_declination", component_property="value"
+            ),
+            prevent_initial_call=True,
+        )(self.write_trigger)
```

Comment:
It looks like this callback is duplicated in `gravity` and `magnetic_vector`. Any chance we can move it to the base class?